weblate/glossary/tasks.py

```diff
@@ -40,6 +47,32 @@ def sync_glossary_languages(pk: int, component: Component | None = None) -> None
         component.create_translations_task()
 
 
+@app.task(trail=False, autoretry_for=(Project.DoesNotExist, WeblateLockTimeoutError))
+def cleanup_stale_glossaries(project: int | Project) -> None:
+    if isinstance(project, int):
+        project = Project.objects.get(pk=project)
+
+    languages_in_non_glossary_components: set[int] = set(
+        Translation.objects.filter(
+            component__project=project, component__is_glossary=False
+        ).values_list("language_id", flat=True)
+    )
+
+    glossary_translations = prefetch_stats(
+        Translation.objects.filter(
+            component__project=project, component__is_glossary=True
+        )
+    )
+    for glossary in glossary_translations:
+        if (
+            glossary.stats.translated == 0
+            and glossary.language_id not in languages_in_non_glossary_components
+        ):
+            glossary.delete()
```

Comment:
It should call remove instead to actually remove the file from the underlying repository (otherwise it will reappear shortly):

```suggestion
            glossary.remove()
```